extensions/GitExtension/FileExplorerGitIntegration/Helpers/GitCommandRunnerResultInfo.cs

```diff
@@ -19,18 +19,22 @@ public class GitCommandRunnerResultInfo
 
     public string? Arguments { get; set; }
 
+    public int? ProcessExitCode { get; set; }
+
     public GitCommandRunnerResultInfo(ProviderOperationStatus status, string? output)
     {
         Status = status;
         Output = output;
     }
 
-    public GitCommandRunnerResultInfo(ProviderOperationStatus status, string? displayMessage, string? diagnosticText, Exception? ex, string? args)
+    public GitCommandRunnerResultInfo(ProviderOperationStatus status, string? output, string? displayMessage, string? diagnosticText, Exception? ex, string? args, int? processExitCode)

```

Comment:
With so many nullable parameters can you split the constructor into multiple constructors so callers can more easily use this object.

I don't like calling constructors with multiple nullable parameters because the constructor does not tell me what combination of values ill throw an exception.

Reply:
Consider adding more constructors instead of more nullable parameters.